internal/handlers/equipment_status.go

```diff
@@ -2,109 +2,369 @@ package handlers
+				WithPayload(buildStringPayload("can't update equipment status on available status"))
 		}
 
-		return eqStatusName.NewGetEquipmentStatusNameOK().WithPayload(&models.SuccessEquipmentStatusNameOperationResponse{
-			Data: mapEquipmentStatusName(foundStatus),
-		})
+		eqStatusResult, err := eqStatusRepository.GetEquipmentStatusByID(
+			ctx, int(*data.ID))
+		if err != nil {
+			c.logger.Error("receiving equipment status by id failed during changing status to available", zap.Error(err))
+			return eqStatus.NewCheckEquipmentStatusDefault(http.StatusInternalServerError).
+				WithPayload(buildStringPayload("can't find equipment status by provided id"))
+		}
+
+		equipmentStatusID := int64(updatedEqStatus.ID)
+		equipmentID := int64(eqStatusResult.Edges.Equipments.ID)
+		return eqStatus.NewUpdateEquipmentStatusOnAvailableOK().WithPayload(
+			&models.EquipmentStatusRepairResponse{
+				Data: &models.EquipmentStatus{
+					ID:          &equipmentStatusID,
+					EndDate:     (*strfmt.DateTime)(&updatedEqStatus.EndDate),
+					StartDate:   (*strfmt.DateTime)(&updatedEqStatus.StartDate),
+					StatusName:  &eqStatusResult.Edges.EquipmentStatusName.Name,
+					EquipmentID: &equipmentID,
+					CreatedAt:   strfmt.DateTime(eqStatusResult.CreatedAt),
+				},
+			})
 	}
 }
 
-func (c EquipmentStatusName) DeleteEquipmentStatusNameFunc(repository domain.EquipmentStatusNameRepository) eqStatusName.DeleteEquipmentStatusNameHandlerFunc {
-	return func(s eqStatusName.DeleteEquipmentStatusNameParams, access interface{}) middleware.Responder {
+func (c EquipmentStatus) PatchEquipmentStatusEditDatesFunc(
+	eqStatusRepository domain.EquipmentStatusRepository,
+) eqStatus.UpdateRepairedEquipmentStatusDatesHandlerFunc {
+	return func(s eqStatus.UpdateRepairedEquipmentStatusDatesParams, access interface{}) middleware.Responder {
 		ctx := s.HTTPRequest.Context()
-		deletedStatus, err := repository.Delete(ctx, int(s.StatusID))
+
+		ok, err := equipmentStatusAccessRights(access)
 		if err != nil {
-			c.logger.Error("delete status failed", zap.Error(err))
-			return eqStatusName.NewDeleteEquipmentStatusNameDefault(http.StatusInternalServerError).
-				WithPayload(buildStringPayload("can't delete status"))
-		}
-		return eqStatusName.NewDeleteEquipmentStatusNameOK().WithPayload(
-			&models.SuccessEquipmentStatusNameOperationResponse{
-				Data: mapEquipmentStatusName(deletedStatus),
-			},
-		)
+			c.logger.Error("Error while getting authorization", zap.Error(err))
+			return orders.NewAddNewOrderStatusDefault(http.StatusInternalServerError).
+				WithPayload(&models.Error{Data: &models.ErrorData{Message: "Can't get authorization"}})
+		}
+
+		if !ok {
+			c.logger.Error("User have no right to update equipment status on available", zap.Any("access", access))
+			return orders.NewAddNewOrderStatusDefault(http.StatusForbidden).
+				WithPayload(&models.Error{Data: &models.ErrorData{Message: "You don't have rights to update equipment status"}})
+		}
```

Comment:
This part seems to be general part of all 3 new methods. Maybe move to function to reduce repetition?